packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -62,11 +62,11 @@ export interface ExportedDayCalendarProps extends ExportedPickersDayProps {
   renderLoading?: () => React.ReactNode;
   /**
    * Formats the day of week displayed in the calendar header.
-   * @param {string} day The day of week provided by the adapter's method `getWeekdays`.
+   * @param {TDate} weekday The day of week provided by the adapter's method `getWeek`.
    * @returns {string} The name to display.
-   * @default (day) => day.charAt(0).toUpperCase()
+   * @default (weekday: TDate) => adapter.format(weekday, 'weekdayShort').charAt(0).toUpperCase()
    */
-  dayOfWeekFormatter?: (day: string) => string;
+  dayOfWeekFormatter?: (weekday: TDate) => string;
```

Comment:
For me we can't do this breaking change during a minor
Sorry if I was not clear, in https://github.com/mui/mui-x/issues/9984#issuecomment-1717634957 I was talking purely about stopping using `adapter.getWeekdays` not being breaking.

If we want to improve `dayOfWeekFormatter`, we have two options:

1. Create a new prop `dayOfWeekFormatterV7` that takes a date, deprecated `dayOfWeekFormatter` and swap in v7
2. Add the date as a 2nd param on `dayOfWeekFormatter` and remove the 1st param in v7.

Reply:
see https://github.com/mui/mui-x/pull/10345#discussion_r1327238451